apps/src/aiTutor/views/AITutorSuggestedPrompts.tsx

```diff
@@ -0,0 +1,115 @@
+import React, {useCallback} from 'react';
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import SuggestedPrompts from '@cdo/apps/aiComponentLibrary/suggestedPrompt/SuggestedPrompts';
+
+import {
+  AITutorTypes as ActionType,
+  AITutorTypesValue,
+} from '@cdo/apps/aiTutor/types';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import {askAITutor} from '../redux/aiTutorRedux';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+
+const QuickActions = {
+  [ActionType.COMPILATION]: "Why doesn't my code compile?",
+  [ActionType.VALIDATION]: "Why aren't my tests passing?",
+};
+
+const AITutorSuggestedPrompts: React.FunctionComponent = () => {
+  const isWaitingForChatResponse = useAppSelector(
+    state => state.aiTutor.isWaitingForChatResponse
+  );
+
+  const level = useAppSelector(state => state.aiTutor.level);
+
+  const sources = useAppSelector(state => state.javalabEditor.sources);
+  const fileMetadata = useAppSelector(
+    state => state.javalabEditor.fileMetadata
+  );
+  const activeTabKey = useAppSelector(
+    state => state.javalabEditor.activeTabKey
+  );
+  const studentCode = sources[fileMetadata[activeTabKey]].text;
+
+  const hasCompilationError = useAppSelector(
+    state => state.javalabEditor.hasCompilationError
+  );
+  const hasRunOrTestedCode = useAppSelector(
+    state => state.javalab.hasRunOrTestedCode
+  );
+  const isRunning = useAppSelector(state => state.javalab.isRunning);
+  const validationPassed = useAppSelector(
+    state => state.javalab.validationPassed
+  );
+
+  const dispatch = useAppDispatch();
+
+  const showCompilationOption =
+    !isRunning &&
+    hasRunOrTestedCode &&
+    hasCompilationError &&
+    !isWaitingForChatResponse;
+  const showValidationOption =
+    hasRunOrTestedCode &&
+    !hasCompilationError &&
+    !validationPassed &&
+    !isWaitingForChatResponse;
+
+  const handleClick = useCallback(
+    (actionType: AITutorTypesValue) => {
+      if (isWaitingForChatResponse) {
+        return;
+      }
+
+      let studentInput = '';
+      let event = '';
+
+      switch (actionType) {
+        case ActionType.COMPILATION:
+          studentInput = QuickActions[ActionType.COMPILATION];
+          event = EVENTS.AI_TUTOR_ASK_ABOUT_COMPILATION;
+          break;
+        case ActionType.VALIDATION:
+          studentInput = QuickActions[ActionType.VALIDATION];
+          event = EVENTS.AI_TUTOR_ASK_ABOUT_VALIDATION;
+          break;
+      }
```

Comment:
This is fine and I'd probably discourage changing it. So this is purely a note.

But.

I always find it frustrating writing code when I need to use a mutable variable because an approach to using an immutable one isn't apparent. Fortunately, I know a pattern that works here, so I figured I'd just note it for future reference. I'm not convinced it adds enough clarity to actually be worth implementing. Unless the fact that mutable variables exist bugs you too. 😄 

```
const actionDispatch = {
  [ActionType.COMPILATION] : {
    studentInput : QuickActions[ActionType.COMPILATION],
    event: EVENTS.AI_TUTOR_ASK_ABOUT_COMPILATION
  },
  [ActionType.VALIDATION] : {
    studentInput : QuickActions[ActionType.VALIDATION],
    event: EVENTS.AI_TUTOR_ASK_ABOUT_VALIDATION
  }
}

const {studentInput, event} = actionDispatch[actionType]
```

One potential advantage to this approach is that it's just data and the `QuickActions` and `EVENTS` objects are already defined externally, so the `actionDispatch` (or whatever it's called) could be defined external to the component and not inlined at all here, which might improve readability.

The counter argument being that it's also moving the lookup farther away, which could be detrimental. So YMMV.

Reply:
Thanks for the info! I'm going to leave as is for now, but appreciate the attention to readability and explanation.